data/data/agent/files/usr/local/bin/load-config-iso.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+set -e
+
+AGENT_CONFIG_ARCHIVE_FILE="config.gz"
+AGENT_CONFIG_MOUNT="/media/config-image"
+CLUSTER_IMAGE_SET="/etc/assisted/manifests/cluster-image-set.yaml"
+
+copy_archive_contents() {
+    tmpdir=$(mktemp --tmpdir -d "config-image--XXXXXXXXXX")
+    cp ${AGENT_CONFIG_MOUNT}/${AGENT_CONFIG_ARCHIVE_FILE} "${tmpdir}"
+    gunzip -f "${tmpdir}"/${AGENT_CONFIG_ARCHIVE_FILE}
+    unzipped_file=$(echo "${tmpdir}/${AGENT_CONFIG_ARCHIVE_FILE}" | cut -d'.' -f1)
+    filelist=$(cpio --list < "${unzipped_file}")
+    echo -e "List of files to copy: ${filelist}"
+
+    # Get the releaseImage in the archive and verify it matches the current cluster-image-set
+    release_image=$(grep releaseImage ${CLUSTER_IMAGE_SET} | sed -n -e 's/^.*releaseImage: //p')
+    arch_image_set=$(cpio -icv --to-stdout ${CLUSTER_IMAGE_SET} < "${unzipped_file}")
+    if [[ $(< "${CLUSTER_IMAGE_SET}") != "${arch_image_set}" ]]; then
+       echo "The cluster-image-set in archive does not match current release ${release_image}"
+       cleanup_files
+       return 1
+    fi
+    echo "Archive on ${devname} contains release ${release_image}"
+
+    # Get array from string
+    IFS=',' read -r -a array <<< "${CONFIG_IMAGE_FILES}"
+
+    # Copy expected files from archive, overwriting the existing file
+    for file in "${array[@]}"
+    do
+       if [[ "${file}" = *.* || "${file}" == "/etc/issue" ]]; then
+	  # Copy file
+          cpio -icvdu "${file}" < "${unzipped_file}"
+          if [[ -f ${file} ]]; then
+             echo "Copied ${file}"
+          else
+             echo "Expected file ${file} is not in archive"
+          fi
+       else
+          # copy all files in directory
+          cpio -icvdu "${file}*" < "${unzipped_file}"
+          if [[ -d ${file} ]]; then
+             echo "Copied files in ${file}"
+          else
+             echo "Could not copy directory ${file}, may be empty, attempting to create it"
+             mkdir -p "${file}"
+          fi
+       fi
+    done
+
+    echo "Successfully copied contents of ${AGENT_CONFIG_ARCHIVE_FILE} on ${devname}"
+
+    # For services that were updated a daemon-reload is required to pick up changes
+    systemctl daemon-reload
```

Comment:
OK, _this_ is the thing I've been told we must never do.
We need to move the variables out of the systemd unit file and into a separate environment file so that the config-drive never updates the systemd unit.

Reply:
I moved the `REQUIRED_MASTER_NODES` and `REQUIRED_WORKER_NODES` to assisted-service.env.template and added that file to list of files to copy. I was able to remove the `systemctl daemon-reload` and confirmed the values are picked up
```
Jun 28 14:20:49 master-0.ostest.test.metalkube.org start-cluster-installation.sh[3237]: Number of required master nodes: 3
Jun 28 14:20:49 master-0.ostest.test.metalkube.org start-cluster-installation.sh[3237]: Number of required worker nodes: 0
Jun 28 14:20:49 master-0.ostest.test.metalkube.org start-cluster-installation.sh[3237]: Total number of required nodes: 3

```